torchvision/transforms/v2/functional/_color.py

```diff
@@ -33,9 +33,14 @@ def rgb_to_grayscale(inpt: torch.Tensor, num_output_channels: int = 1) -> torch.
 def _rgb_to_grayscale_image(
     image: torch.Tensor, num_output_channels: int = 1, preserve_dtype: bool = True
 ) -> torch.Tensor:
-    if image.shape[-3] == 1:
+    if num_output_channels not in (1, 3):
+        raise ValueError(f"num_output_channels must be 1 or 3, got {num_output_channels}.")
```

Comment:
This check is already done in all of the call sites of this helper, so it's not needed here. In general I agree we should error at the lower-level functions like this one instead of earlier. Maybe it's be better to error here instead of the callsites, but since it would require more change than necessary for this feature, let's leave it out for now.

Reply:
Replaced with a TODO. Let me know if something else is needed here.